hazelcast/src/main/java/com/hazelcast/map/impl/proxy/NearCachedMapProxyImpl.java

```diff
@@ -262,6 +262,12 @@ protected Data removeInternal(Data key) {
     }
 
     @Override
+    protected void removeAllInternal(Predicate predicate) {
+        super.removeAllInternal(predicate);
+        nearCache.clear();
```

Comment:
can we evaluate the predicate locally (on entries in the near-cache)? is it too silly? :)